spec/helpers/http_mock.lua

```diff
@@ -25,6 +25,32 @@ for _, module in ipairs(modules) do
   end
 end
 
+-- get a session from the logs with a timeout
+-- throws error if no request is recieved within the timeout
+-- @treturn table the session
+function http_mock:get_session()
+  local ret
+  self.eventually:has_session_satisfy(function(s)
+    ret = s
+    return true
+  end)
+  return ret
+end
+
+-- get a request from the logs with a timeout
+-- throws error if no request is recieved within the timeout
+-- @treturn table the request
+function http_mock:get_request()
+  return self:get_session().req
```

Comment:
should we `get` the session object first and then access `.req` only if it's  returned correctly? Or are we sure that if it isn't valid we are already handling with an error elsewhere?

Reply:
The `get_session` throws an error if no session is got in time.